hysds/utils.py

```diff
@@ -343,15 +349,29 @@ def get_payload_hash(payload):
     ).hexdigest()
 
 
+def no_dedup_job(details):
+    logger.info("Giving up querying for dedup jobs with args {args} and kwargs {kwargs}".format(**details))
+    return None
+
+
 @backoff.on_exception(
     backoff.expo, requests.exceptions.RequestException, max_tries=8, max_value=32
 )
+@backoff.on_exception(
+    backoff.expo, NoDedupJobFoundException, max_tries=8, max_value=32, on_giveup=no_dedup_job
```

Comment:
My understanding of the exponential backoff is that multiple decorators are supported (https://pypi.org/project/backoff/):

![image](https://user-images.githubusercontent.com/42812746/109040111-6c50e900-7682-11eb-96b0-6ec07d473985.png)

Given this, it seems like the decorators here will handle both exceptions? Let me know if you think otherwise. 
With the `on_giveup` function, during testing I discovered that it will still re-raise the exception back to the caller even though the `no_dedup_job` returns None. That's the reason why the orchestrator and the job_worker needed to be updated as well to surround the `query_dedup_job` function with a try/except block to catch the NoDedupJobFoundException to essentially return None. If we decide to move forward with what you've suggested and combine the 2 exceptions into 1 decorator, I think the `on_giveup` function can be left as-is. Thoughts?

Reply:
@mcayanan: I like that approach better then. So the original decorator for handling `requests.exceptions.RequestException` should be restored and a second decorator handling the `NoDedupJobFoundException` added as you have here. Did I get that right? If so, let's go for it.

Reply:
Cool. I think that's how it was implemented initially? Here's the change:

https://github.com/hysds/hysds/blob/ff62cd5e497b3c3b5117f423e3ae31b34184a269/hysds/utils.py#L357-L363